script/testing/micro_bench/google_benchmark/gbench_run_result.py

```diff
@@ -0,0 +1,66 @@
+import json
+import os
+from datetime import datetime
+
+from micro_bench.google_benchmark.gbench_test_result import GBenchTestResult
+from util.constants import LOG
+
+
+class GBenchRunResult(object):
+    """ The results from running a Google benchmark file (i.e. bwtree_benchmark)
+        This may contain multiple individual benchmark tests. This class will
+        parse the results and create a map of benchmark test results where the
+        key is (suite_name, test_name) and the value is a GBenchTestResult.
+    """
+
+    def __init__(self, json_results):
+        self.results = json_results
+        self.benchmarks = self._populate_benchmark_results(self.results)
+        self.benchmark_name = self._get_benchmark_name()
+
+    def _populate_benchmark_results(self, results_dict):
+        """ populates the benchmark map with GBenchTestResult objects and adds
+            a timestamp from the higher level JSON into the individual test
+            result
+        """
+        benchmarks = {}
+        for bench in results_dict.get('benchmarks', []):
+            bench['timestamp'] = self._get_datetime()
+            test_result = GBenchTestResult(bench)
+            key = (test_result.suite_name, test_result.test_name)
+            benchmarks[key] = test_result
```

Comment:
Should it be `for bench in results_dict.get('benchmarks', {}):`? Because the `bench`is a dict, not a list by default.

Reply:
good catch

Reply:
oh wait actually no we want to default value to be an array cause `results_dict['benchmarks']` is an array. Therefore we would return an empty array and not do any iteration.

Reply:
Ah silly me! You are correct!